### ISSUE TITLE
only single action supported

### ISSUE THREAD

**Issue by mattinjersey** [tensorflow/agents#112]
Getting this error...is this my mistake or do some agents only support a single action?

ValueError: Only a single action is supported by this network
  In call to configurable 'CriticNetwork' (<function _NetworkMeta.__new__.<locals>.capture_init at 0x00000203ED9DEEA0>)
  In call to configurable 'train_eval' (<function train_eval at 0x00000203EE878BF8>)

**Comment by sguada** [comment]
You need to define a network that can support multiple Actions, you can use CriticNetwork as an example.

**Comment by mattinjersey** [comment]
Well for example with td3 agent, I define Critic Network as follows, and the action spec defined with 2 actions, and I get such error. The only agent that worked was PPOAgent. 
Are we sure that all the agents can handle multiple actions?

_critic_net_input_specs = (tf_env.time_step_spec().observation,
                              tf_env.action_spec())

    critic_net = critic_network.CriticNetwork(
        critic_net_input_specs,
        observation_fc_layer_params=critic_obs_fc_layers,
        action_fc_layer_params=critic_action_fc_layers,
        joint_fc_layer_params=critic_joint_fc_layers,
    )_

**Comment by mattinjersey** [comment]
Right, I see in ddpg that critic_network can only handle 1 observation.
So it seems like ddpg agent is limited to 1 observation.

**Comment by mattinjersey** [comment]
SAC only limited to 1 observation, it appears.

**Comment by sguada** [comment]
So would need to write your own ActorNetwork and your own CriticNetwork classes that can handle multiple observations and multiple actions.

**Comment by mattinjersey** [comment]
ok I may try it. I'm not sure why the PPO Agent is written so that it can accept arbitrary number of actions/observations but the CriticNetwork is written differently.

**Comment by basvanopheusden** [comment]
I am also struggling with the same issue, @sguada, if you could give me some pointers as to how to create custom ActorDistributionNetworks, I would really appreciate it!

**Comment by liujuncn** [comment]
@sguada if I want to create my own ActorNetwork/CriticNetwork classes to support multiple observations(spec defined in a dict), which classes and their methods are involved to override?
